Networking/NetworkingTests/Remote/ProductsRemoteTests.swift

```diff
@@ -259,11 +259,13 @@ class ProductsRemoteTests: XCTestCase {
         let productName = "This is my new product name!"
         let productDescription = "Learn something!"
         let product = sampleProduct()
-        remote.updateProduct(product: product) { (product, error) in
-            XCTAssertNil(error)
-            XCTAssertNotNil(product)
-            XCTAssertEqual(product?.name, productName)
-            XCTAssertEqual(product?.fullDescription, productDescription)
+        remote.updateProduct(product: product) { result in
+            guard case let .success(product) = result else {
+                XCTFail("Unexpected result: \(result)")
+                return
+            }
+            XCTAssertEqual(product.name, productName)
+            XCTAssertEqual(product.fullDescription, productDescription)
             expectation.fulfill()
```

Comment:
As part of the refactoring, we can avoid the expectation declared outside, and use the callback
```
waitForExpectation { expectation in
expectation.fulfill()
}
```

Reply:
Great suggestion! updated in https://github.com/woocommerce/woocommerce-ios/pull/2533/commits/9f6c5b1d6666bbd75ba2fa8564cbd4bcd2bd93a7